scalpel-core/src/Text/HTML/Scalpel/Internal/Scrape.hs

```diff
@@ -39,50 +39,49 @@ import Control.Monad.Writer (MonadWriter)
 import Control.Monad.Fix
 import Data.Functor.Identity
 import Data.Maybe
+import Data.Text (Text)
 
 import qualified Control.Monad.Fail as Fail
 import qualified Data.Vector as Vector
-import qualified Text.HTML.TagSoup as TagSoup
-import qualified Text.StringLike as TagSoup
-
+import qualified Data.Text as T
+import qualified Data.Text.Lazy as TL
+import qualified Text.HTML.Parser as HP
 
 -- | A 'ScraperT' operates like 'Scraper' but also acts as a monad transformer.
-newtype ScraperT str m a = MkScraper (ReaderT (TagSpec str) (MaybeT m) a)
+newtype ScraperT m a = MkScraper (ReaderT (TagSpec) (MaybeT m) a)
     deriving (Functor, Applicative, Alternative, Monad, MonadPlus, MonadFix,
               MonadIO, MonadCont, MonadError e, MonadState s, MonadWriter w)
 
 #if MIN_VERSION_base(4,9,0)
-deriving instance Monad m => Fail.MonadFail (ScraperT str m)
+deriving instance Monad m => Fail.MonadFail (ScraperT m)
 #else
-instance Fail.MonadFail m => Fail.MonadFail (ScraperT str m) where
+instance Fail.MonadFail m => Fail.MonadFail (ScraperT m) where
   fail = lift . Fail.fail
 #endif
 
-instance MonadTrans (ScraperT str) where
+instance MonadTrans (ScraperT) where
   lift op = MkScraper . lift . lift $ op
 
-instance MonadReader s m => MonadReader s (ScraperT str m) where
+instance MonadReader s m => MonadReader s (ScraperT m) where
   ask = MkScraper (lift . lift $ ask)
   local f (MkScraper op) = (fmap MkScraper . mapReaderT . local) f op
 
 -- | A value of 'Scraper' @a@ defines a web scraper that is capable of consuming
--- a list of 'TagSoup.Tag's and optionally producing a value of type @a@.
-type Scraper str = ScraperT str Identity
+-- a list of 'HP.Tag's and optionally producing a value of type @a@.
```

Comment:
'HP.Tag' should be 'HP.Token'? See a couple instances throughout.

Reply:
Resolved with https://github.com/fimad/scalpel/pull/115/commits/69b57d43b7ea2ddd2690ff59d160f47a35284daa.
(Remaining matches are entities from `HP` that is `Text.HTML.Parser itself`.)